### PR TITLE
I like guard.

### PR DESCRIPTION
**Pull request by kwerle**
This way you can leave guard running [in a docker container] while doing development work.  Makes writing and using tests very convenient, in my opinion.

Oops - close this by accident the first time?

### CODE REVIEW

File: Dockerfile.test [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# docker build -t lsr-test -f Dockerfile.test .
```
Comment (mtsmfm) [inline Dockerfile.test]: Why don't you invoke guard via docker-compose?

https://github.com/mtsmfm/language_server-ruby/tree/ab8998db3e14a414ef51130611a743c39978db8f#run-test

We don't have to create new dockerfile

File: Guardfile [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# A sample Guardfile
```
Comment (mtsmfm) [inline Guardfile]: Could you remove comments?

File: Dockerfile.test (hunk 1 above)
Comment (kwerle) [reply Dockerfile.test]: Eh.  
Composition just feels heavy to me.  And not all platforms have docker-compose by default.

Which compose file do you suggest I use, and how would I invoke it?

File: Guardfile (hunk 2 above)
Comment (kwerle) [reply Guardfile]: Will do.

File: Dockerfile.test (hunk 1 above)
Comment (mtsmfm) [reply Dockerfile.test]: If you're using Docker for Mac, docker-compose is already installed.

You can run any commands via docker-compose.
So in this case, `docker-compose run app bundle exec guard` is what you want.
If you'd like to use old ruby, run `docker-compose run ruby-2-3 bundle exec guard`.

There are three docker-compose related files:

- [docker-compose.yml](https://github.com/mtsmfm/language_server-ruby/blob/ab8998db3e14a414ef51130611a743c39978db8f/docker-compose.yml)
- [docker-compose.override.yml](https://github.com/mtsmfm/language_server-ruby/blob/ab8998db3e14a414ef51130611a743c39978db8f/docker-compose.override.yml)
- [docker-compose.ci.yml](https://github.com/mtsmfm/language_server-ruby/blob/ab8998db3e14a414ef51130611a743c39978db8f/docker-compose.ci.yml)

docker-compose will use docker-compose.yml and docker-comopse.override.yml automatically.
docker-compose.ci.yml is only for CI and it isn't used on development env.

File: language_server.gemspec [hunk 3]
```diff
@@ -31,6 +31,9 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency "bundler", "~> 1.14"
   spec.add_development_dependency "rake", "~> 10.0"
   spec.add_development_dependency "minitest", "~> 5.0"
+  spec.add_development_dependency "guard"
+  spec.add_development_dependency "guard-minitest"
+  spec.add_development_dependency "minitest-color"
```
Comment (mtsmfm) [inline language_server.gemspec]: [minor] I think adding minitest-color isn't related to this PR.
It's ok at this time but please split PR at next time 🙏

**Comment by kwerle** [comment]
I'm afraid I don't know of a way.  I never really got into system notifications even before docker.

File: test/language_server/project/node_test.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
+require 'test_helper'
+
+module LanguageServer
+  class Project
+    class NodeTest < Minitest::Test
```
Comment (mtsmfm) [inline test/language_server/project/node_test.rb]: Please don't add test in this PR.
This PR should contain only guard related things.

**Comment by mtsmfm** [comment]
> I'm afraid I don't know of a way.

Don't worry, it's OK. I'm just curious about that.

**Comment by mtsmfm** [comment]
Thanks!

**Pull request merged**